### PR TITLE
chore(db): Split DB and state for removing `Group.score` column

### PR DESCRIPTION
**Pull request by leeandher**
Follow-up to https://github.com/getsentry/sentry/pull/78769

Splits the DB and state, removing the `Group.score` column from Django.
Also removes the last use of it I could find, where a value is stored in the `save()` method.

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
It might also need to be removed here? 

https://github.com/getsentry/sentry/blob/9ab0573e22d9b45c98bef4c5000abd85f316fff2/src/sentry/testutils/cases.py#L1477

You also need to remove the column from the model

**Comment by leeandher** [comment]
> It might also need to be removed here?
> 
> https://github.com/getsentry/sentry/blob/9ab0573e22d9b45c98bef4c5000abd85f316fff2/src/sentry/testutils/cases.py#L1477
> 
> You also need to remove the column from the model

Thanks! Just realized my search filtered test files, good catch 🙏. TY on mentioning removing the column from the model as well, would have missed that.

File: src/sentry/testutils/cases.py [hunk 1]
```diff
@@ -1474,7 +1474,6 @@ def __wrap_group(self, group):
                 self.to_snuba_time_format(group.last_seen),
                 self.to_snuba_time_format(group.first_seen),
                 group.data,
-                group.score,
```
Comment (wedamija) [inline src/sentry/testutils/cases.py:1477]: I think you will also need to remove `score` from the list of strings above

**Review by wedamija — APPROVED** [review]

File: src/sentry/models/group.py [hunk 2]
```diff
@@ -624,9 +623,6 @@ def save(self, *args, **kwargs):
             self.message = truncatechars(self.message.splitlines()[0], 255)
         if self.times_seen is None:
             self.times_seen = 1
-        self.score = type(self).calculate_score(
-            times_seen=self.times_seen, last_seen=self.last_seen
-        )
```
Comment (wedamija) [inline src/sentry/models/group.py:629]: Let's also remove the `calculate_score` method?

Comment (leeandher) [reply src/sentry/models/group.py:629]: removed it and `get_score` as well. I was also able to remove `ScoreClause` from `src/sentry/event_manager.py`which I believe is unused as well.

File: src/sentry/buffer/base.py [hunk 3]
```diff
@@ -162,12 +161,6 @@ def process(
             # HACK(dcramer): this is gross, but we don't have a good hook to compute this property today
             # XXX(dcramer): remove once we can replace 'priority' with something reasonable via Snuba
             if model is Group:
-                if "last_seen" in update_kwargs and "times_seen" in update_kwargs:
-                    update_kwargs["score"] = ScoreClause(
-                        group=None,
-                        times_seen=update_kwargs["times_seen"],
-                        last_seen=update_kwargs["last_seen"],
-                    )
```
Comment (markstory) [inline src/sentry/buffer/base.py:170]: :clap: :clap:

**Review by wedamija — APPROVED** [review]

**Pull request merged**